src/main/java/com/minecolonies/coremod/entity/ai/citizen/school/EntityAIWorkTeacher.java

```diff
@@ -0,0 +1,214 @@
+          new AITarget(TEACH, this::teach, TICKS_SECOND)
+        );
+        worker.setCanPickUpLoot(true);
+    }
+
+    /**
+     * Decide what to do next.
+     * @return the next state to go to.
+     */
+    private IAIState decide()
+    {
+        final int paperInBuilding = InventoryUtils.getItemCountInProvider(getOwnBuilding(), PAPER);
+        final int paperInInv = InventoryUtils.getItemCountInItemHandler((worker.getInventoryCitizen()), PAPER);
+        if (paperInBuilding + paperInInv <= 0)
+        {
+            requestPaper();
+        }
+
+        if (paperInInv == 0 && paperInBuilding > 0)
+        {
+            needsCurrently = PAPER;
+            return GATHERING_REQUIRED_MATERIALS;
+        }
+
+        final List<? extends AbstractEntityCitizen> pupils = world.getEntitiesWithinAABB(ModEntities.CITIZEN, getTargetableArea(), cit -> cit.isChild() && cit.ridingEntity != null && cit.getCitizenJobHandler().getColonyJob() instanceof JobPupil);
+        if (pupils.size() > 0)
+        {
+            pupilToTeach = pupils.get(worker.getRandom().nextInt(pupils.size()));
+            return TEACH;
+        }
+
+        return START_WORKING;
+    }
+
+
+    private IAIState teach()
+    {
+        if (pupilToTeach == null || pupilToTeach.ridingEntity == null)
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(pupilToTeach.getPosition()))
+        {
+            return getState();
+        }
+
+        if (maxSittingTicks == 0 || worker.ridingEntity == null)
+        {
+            // Sit for 2-100 seconds.
+            final int jobModifier = 100 / worker.getCitizenData().getJobModifier();
+            maxSittingTicks = worker.getRandom().nextInt(jobModifier / 2) + jobModifier / 2;
+
+            final SittingEntity entity = (SittingEntity) ModEntities.SITTINGENTITY.create(world);
+            entity.setPosition(worker.posX, worker.posY - 1f, worker.posZ);
+            entity.setMaxLifeTime(maxSittingTicks * 20);
+            world.addEntity(entity);
+            worker.startRiding(entity);
+            worker.getNavigator().clearPath();
+        }
+
+        sittingTicks++;
+        if (sittingTicks < maxSittingTicks)
+        {
+            return getState();
+        }
+
+        if (worker.ridingEntity != null)
+        {
+            worker.stopRiding();
+            worker.setPosition(worker.posX, worker.posY + 1, worker.posZ);
+        }
+
+        final int slot = InventoryUtils.findFirstSlotInItemHandlerWith(worker.getInventoryCitizen(), PAPER);
+        if (slot != -1)
+        {
+            InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoNextFreeSlotInItemHandler(
+              worker.getInventoryCitizen(),
+              PAPER,
+              1, pupilToTeach.getInventoryCitizen()
+            );
+        }
+
+        pupilToTeach.getCitizenData().getCitizenSkillHandler().addXpToSkill(Skill.Intelligence, 1, pupilToTeach.getCitizenData());
```

Comment:
how much is 1 Intelligence xp per second over a half day of school in int levels? not sure if its enough

Reply:
added the bonus to the wrong one. (Paper consumption gives big bonus too)